crates/uv-client/src/cached_client.rs

```diff
@@ -519,6 +563,103 @@ impl CachedClient {
         };
         Ok((response, cache_policy))
     }
+
+    /// Perform a [`CachedClient::get_serde`] request with a default retry strategy.
+    #[instrument(skip_all)]
+    pub async fn get_serde_with_retry<
+        Payload: Serialize + DeserializeOwned + 'static,
+        CallBackError: std::error::Error + 'static,
+        Callback,
+        CallbackReturn,
+    >(
+        &self,
+        req: Request,
+        cache_entry: &CacheEntry,
+        cache_control: CacheControl,
+        response_callback: Callback,
+    ) -> Result<Payload, CachedClientError<CallBackError>>
+    where
+        Callback: Fn(Response) -> CallbackReturn,
+        CallbackReturn: Future<Output = Result<Payload, CallBackError>>,
+    {
+        let payload = self
+            .get_cacheable_with_retry(req, cache_entry, cache_control, |resp| async {
+                let payload = response_callback(resp).await?;
+                Ok(SerdeCacheable { inner: payload })
+            })
+            .await?;
+        Ok(payload)
+    }
+
+    /// Perform a [`CachedClient::get_cacheable`] request with a default retry strategy.
+    #[instrument(skip_all)]
+    pub async fn get_cacheable_with_retry<
+        Payload: Cacheable,
+        CallBackError: std::error::Error + 'static,
+        Callback,
+        CallbackReturn,
+    >(
+        &self,
+        req: Request,
+        cache_entry: &CacheEntry,
+        cache_control: CacheControl,
+        response_callback: Callback,
+    ) -> Result<Payload::Target, CachedClientError<CallBackError>>
+    where
+        Callback: Fn(Response) -> CallbackReturn,
+        CallbackReturn: Future<Output = Result<Payload, CallBackError>>,
+    {
+        let mut attempt = 0;
+        loop {
+            let fresh_req = req.try_clone().expect("HTTP request must be cloneable");
+            let result = self
+                .get_cacheable(fresh_req, cache_entry, cache_control, &response_callback)
+                .await;
```

Comment:
Should there be some delay here?

Reply:
Oh, I guess? I can try to reuse the retry policy from the middleware.

Reply:
I would definitely expect an exponential backoff.

Reply:
Done, good call, thank you.